File: project/6.EX_総合理解度チェック/py/py_ex6_ex_3/src/ex6_ex_3.py
'''
Created on 2023/05/13

@author: Keroichi.T
'''
from abc import ABC, abstractmethod


class Monster( ABC ):
    @abstractmethod
    def display(self):
        pass

class Vampire( Monster ):
    def display(self):
        print( self.__class__.__name__ )
    
class Werewolf( Monster ):
    def display(self):
        print( "Werewolf" )
    
class Zombie( Monster ):
    def display(self):
        print( "Zombie" )
    
def main():

    # 繰返し変数が必要ないときは、「_」や「dummy」が使われる
    # これらを使うと、未使用変数の警告も出ないようになる
    mon = []
    for _ in range( 2 ):
        mon.append( Vampire() )
        mon.append( Werewolf() )
        mon.append( Zombie() )

    for m in mon:
        m.display()

if __name__ == '__main__':
    main()

File: project/6.EX_総合理解度チェック/java/java_ex6_ex_3/src/java_ex6_ex_3/Ex6_ex_3.java
package java_ex6_ex_3;

abstract class Monster {
	void display() {}
}

class Vampire extends Monster {
	void display() {
		System.out.println( "Vampire" );
	}
}

class Werewolf extends Monster {
	void display() {
		System.out.println( "Werewolf" );
	}
}

class Zombie extends Monster {
	void display() {
		System.out.println( "Zombie" );
	}
}


public class Ex6_ex_3 {

	public static void main(String[] args) {
		Monster[] mon = new Monster[6];
		
		for( int i = 0; i < 6; i += 3 ) {
			mon[ i + 0 ] = new Vampire();
			mon[ i + 1 ] = new Werewolf();
			mon[ i + 2 ] = new Zombie();
		}
		
		for( int i = 0; i < 6; i++ ) {
			mon[ i ].display();
		}
	}

}
